packages/legacy-workbench/src/js/angular/security/controllers/user-settings.controller.js

```diff
@@ -138,8 +146,8 @@ function UserSettingsController($scope, toastr, $window, $timeout, $jwtAuth, $ro
             .then(() => {
                 toastr.success($translate.instant('security.user.updated', {name: $scope.user.username}));
                 $scope.updateCurrentUserData();
-                ThemeService.toggleThemeMode($scope.selectedThemeMode);
-                WorkbenchSettingsStorageService.saveWorkbenchSettings($scope.workbenchSettings);
+                themeService.setThemeMode($scope.selectedThemeMode);
+                applicationSettingsStorageService.setApplicationSettings($scope.workbenchSettings);
```

Comment:
IMO, this persistence should be handled in ThemeService. Otherwise, if we change the theme somewhere else, we need to know that it isn't persisted automatically and we have to handle it ourselves, for example, when automatically switching between dark and light mode based on the OS setting.